src/Sarif.Multitool/Rules/RuleResources.resx

```diff
@@ -210,6 +210,18 @@
   <data name="SARIF1019_RuleIdMustBePresentAndConsistent" xml:space="preserve">
     <value>In every result, at least one of the properties result.ruleId and result.rule.id must be present. If both are present, they must be equal.</value>
   </data>
+  <data name="SARIF1022_PreferMessageIdInsteadOfMessageText" xml:space="preserve">
+    <value>Prefer message.id instead of message.text</value>
+  </data>
+  <data name="SARIF1022_ResultShouldUseMessageIdInsteadOfMessageText" xml:space="preserve">
+    <value>{0}: The result should use message.id instead of message.text value '{1}'</value>
+  </data>
+  <data name="SARIF1022_RuleMetadataDoesntIncludeMessageId" xml:space="preserve">
+    <value>{0}: The rule metadata doesn't include the message id value '{1}'</value>
```

Comment:
>The rule metadata doesn't include the message id value '{1}' [](start = 16, length = 60)

{0}: The result message specifies the id value `{1}`, but the metadata for rule `{2}` does not contain a message string with that id.

{2} should be `result.ruleId` or `result.rule.id` (whichever one you find first).

The SDK should have a helper method on the `Result` object to get `RuleId` or `Rule.Id`.

Reply:
{0} = pointer
{1} = result.message.id
{2} = reuslt.ruleid ?? result.rule.id